src/actions/cartListActions.ts

```diff
@@ -76,6 +76,75 @@ export const addCartList = async (recipeId: string, ingredientId: number): Promi
   }
 };
 
+export const removeCartList = async (recipeId: string, cartListItemId: number) => {
+  const supabaseServerClient = createServerActionClient<Database>({ cookies });
+
+  const {
+    data: { session },
+  } = await supabaseServerClient.auth.getSession();
+
+  if (!session) redirect("/login");
+
+  try {
+    const cartList = await prisma.cartList.findFirst({
+      where: {
+        recipeId,
+        userId: session.user.id,
+      },
+      include: {
+        CartListItem: true,
+      },
+    });
+
+    const isNotfoundRecipeInCartList = cartList === null;
+
+    if (isNotfoundRecipeInCartList) {
+      return {
+        isSuccess: false,
+        error: "このアイテムは既に削除されています。ページを更新するか、しばらくたってから再度アクセスしてください。",
+      };
+    }
+    const cartListItemSize = await prisma.cartListItem.count({
+      where: {
+        cartListId: cartList.id,
+      },
+    });
+
+    if (cartListItemSize === 1) {
+      // 対象のレシピに紐づくアイテムが1つしかない場合はレシピも削除する
+      await prisma.$transaction([
+        prisma.cartListItem.delete({
+          where: {
+            id: cartListItemId,
+          },
+        }),
+        prisma.cartList.delete({
+          where: {
+            id: cartList.id,
+          },
+        }),
+      ]);
```

Comment:
こちらはスキーマにカスケードDelete を設定しているので
https://github.com/qin-team-recipe/12-recipe-app/blob/0891195e1c92ff7636e940c298c75ca0b00c87a4/src/prisma/schema.prisma#L227

cartListのみの削除でいいと思います！

```suggestion
  await prisma.cartList.delete({
    where: {
      id: cartList.id,
    },
  });
```